ctest.h

```diff
@@ -22,6 +22,41 @@
 #define CTEST_IMPL_FORMAT_PRINTF(a, b)
 #endif
 
+/* To enforce a total ordering on tests defined within a file, we can use the
+ * non-C-standard-but-widely-implemented `__COUNTER__`.  This way, the tests
+ * may be sorted in program definition order.  If `__COUNTER__` isn't available,
+ * the C-standard `__LINE__` will have the same effect _most_ of the time.
+ *
+ * Notably, this fails when multiple `CTEST` expansions occur on the same line.
+ * That might seem like an insane thing to do, but could easily happen if
+ * multiple `CTEST` invocations occur in a macro.  With `__LINE__`, the ordering
+ * of colinear test cases is not defined.  Oh well...  */
+#if defined(__COUNTER__)
+#define CTEST_IMPL_COUNTER __COUNTER__
+#else
+#define CTEST_IMPL_COUNTER __LINE__
+#endif
+
+/* gcc and clang implement preprocessor-based hardening that replaces certain
+ * pointer-problem-prone library functions with new ones that take additional
+ * argument(s) if information about object size is available at compile-time.
+ * Since we build a linker set out of statically-defined objects, we treat
+ * pointers into the linker set as pointers into an ordinary array, but the
+ * compiler has no idea that the memory is contiguous, so hardened library
+ * functions that receive a pointer into the linker set will assert a buffer
+ * overrun error if that function accesses past the end of the first item.
+ * Therefore, we need to trick the compiler into not invoking these hardened
+ * functions when we are passing them a pointer into the linker set.
+ *
+ * Luckily, since the mechanism is preprocessor-based, the solution is simple:
+ * wrap the function name in parens so that e.g. memcpy does not become
+ * __memcpy_chk during preprocessing.  */
+#if _FORTIFY_SOURCE > 0
```

Comment:
Check that `_FORTIFY_SOURCE` is defined before using it. Otherwise warning will be emitted if `-Wundef` is used.

Reply:
On a second thought, why don't we just always wrap the function names in braces where non-macro is required?..

Reply:
Yes, always wrapping in parens is simpler and avoids a possible `-Wundef` warning -- I have pushed an update.